src/common/utils/validate.go

```diff
@@ -0,0 +1,24 @@
+package utils
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const nameComponent = `[a-z0-9]+((?:[._]|__|[-]*)[a-z0-9]+)*`
+
+// TagRegexp is regular expression to match image tags, for example, 'v1.0'
+var TagRegexp = regexp.MustCompile(`^[\w][\w.-]{0,127}$`)
+
+// RepoRegexp is regular expression to match repo name, for example, 'busybox', 'stage/busybox'
+var RepoRegexp = regexp.MustCompile(fmt.Sprintf("^%s(/%s)*$", nameComponent, nameComponent))
```

Comment:
according to the docker official document, https://forums.docker.com/t/docker-registry-v2-spec-and-repository-naming-rule/5466

 If a repository name has two or more path components, they must be separated by a forward slash ("/").

Why do we only support two path components?

Reply:
You may misunderstand it, it do support multiple components.